demos/jobs/jobs_demo_mosquitto/jobs_demo_mosquitto.c

```diff
@@ -0,0 +1,1305 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * This demonstration downloads files from URLs present in job documents
+ * received from the AWS IoT Jobs service. It shows the use of the jobs
+ * library with the Mosquitto client MQTT library for communicating with the
+ * AWS IoT Jobs service.  More details are available in the usage function
+ * in this file.  Note: This demo focuses on use of the jobs library;
+ * a thorough explanation of libmosquitto is beyond the scope of the demo.
+ */
+
+/* C standard includes. */
+#include <assert.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+/* POSIX includes. */
+#include <signal.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <err.h>
+#include <getopt.h>
+
+#include <mosquitto.h>
+#if ( LIBMOSQUITTO_VERSION_NUMBER < 1004010 )
+    #error Please use libmosquitto at version 1.4.10 or higher.
+#endif
+
+#include "demo_config.h"
+#include "jobs.h"
+#include "core_json.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief MQTT server port number.
+ *
+ * AWS IoT Core uses this port for MQTT over TLS.
+ */
+#define DEFAULT_MQTT_PORT       ( 8883 )
+
+/**
+ * @brief Certificate Authority Directory.
+ *
+ * Debian and Ubuntu use this directory for CA certificates.
+ */
+#define DEFAULT_CA_DIRECTORY    "/etc/ssl/certs"
+
+/**
+ * @brief ALPN (Application-Layer Protocol Negotiation) name for AWS IoT MQTT.
+ */
+#define ALPN_NAME               "x-amzn-mqtt-ca"
+
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Describe program usage on stderr.
+ *
+ * @param[in] programName the value of argv[0]
+ */
+static void usage( const char * programName )
+{
+    fprintf( stderr,
+             "\nThis demonstration downloads files from URLs received via AWS IoT Jobs.\n"
+             "(See https://docs.aws.amazon.com/iot/latest/developerguide/iot-jobs.html for an introduction.)\n"
+             "\nCreating a job may be done with the AWS console, or the aws cli, e.g.,\n"
+             "$ aws iot create-job --job-id t12 --targets arn:aws:iot:us-east-1:1234567890:thing/device1 \\\n"
+             "  --document '{\"url\":\"https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.8.5.tar.xz\"}'\n"
+             "\nTo execute the job, on the target device run the downloader with the target's credentials, e.g.,\n"
```

Comment:
Make reader's experience better!
```suggestion
             "\nTo execute the job, on the target device run the demo program with the device's credentials, e.g.,\n"
```

Reply:
changed